packages/docusaurus/src/client/serverEntry.js

```diff
@@ -28,6 +28,23 @@ import {
 } from './LinksCollector';
 import ssrTemplate from './templates/ssr.html.template';
 
+// eslint-disable-next-line no-restricted-imports
+import {memoize} from 'lodash';
+
+const customEtaConfig = {
+  name: 'ssr-template',
+  rmWhitespace: true,
+};
+
+const getCompiledSSRTemplate = memoize(() => {
+  return eta.compile(ssrTemplate.trim(), customEtaConfig);
+});
+
+function renderSSRTemplate(data) {
+  const compiled = getCompiledSSRTemplate();
+  return compiled(data, {...eta.defaultConfig, ...customEtaConfig});
```

Comment:
@nebrelbug I found this part a bit confusing. Why do I need to provide some config in the compiled template. Can't it uses the partial config I set during compilation?

Reply:
also this looks weird to me to expect a full config here, why not partial + you merge to defaults in the lib directly?

Reply:
Hi @slorber, good question. You need to provide a config to the compiled template function because the config object holds partials, filters, and other utility methods Eta uses while actually rendering the template.

Reply:
We do accept partial configs in `compile` and `render`. Those functions merge passed-in config options with Eta's default configuration.

It's better for performance, though, to not have to merge the config every time we call the compiled function. Plus, in most cases it would be redundant (most users call `render()` to render a template, which already merges the config before calling the compiled template function and returning the result) :+1:

Reply:
I did realize that in this case, we can just call the compiled function with `eta.defaultConfig`. The `{rmWhitespace: true}` option is applied only when the template is compiled, and doesn't affect the template's rendering.